src/main/java/duke/parser/Parser.java

```diff
@@ -0,0 +1,154 @@
+package duke.parser;
+
+import duke.command.*;
+import duke.exception.*;
+import duke.task.*;
+
+import java.util.ArrayList;
+import java.util.regex.Pattern;
+
+/**
+ * A class that will make sense of the user's commands through string parsing
+ */
+public class Parser {
+    private static final Pattern pattern = Pattern.compile("-?\\d+(\\.\\d+)?");
+    private static TaskList tasks;
+
+    /**
+     * Decides which parsing is needed for the user input
+     *
+     * @param input the input keyed in by user
+     * @return the type of command the input is
+     * @throws DukeException if the users enter a wrong input
+     */
+    public static Command parse(String input, TaskList tl) throws DukeException{
+        tasks = tl;
+        if (input.equals("list")) {
+            return new ListCommand();
+        } else if (input.equals("bye")) {
+            return new ByeCommand();
+        } else if (input.startsWith("done")) {
+            int index = parseDone(input);
+            return new DoneCommand(index);
+        } else if (input.startsWith("delete")) {
+            int index = parseDelete(input);
+            return new DeleteCommand(index);
+        } else if (input.startsWith("find")) {
+            String target = parseFind(input);
+            return new FindCommand(target);
+        } else {
+            if (!input.startsWith("todo") && !input.startsWith("deadline") && !input.startsWith("event")) {
```

Comment:
This is double checking for input validity! Would be cleaner to check for invalid command input at the end of this else block e.g. if else statements checking if it starts with "todo"/"event"/"deadline", and one last else that throws a DukeException otherwise.